model-optimizer/mo/front/common/partial_infer/utils.py

```diff
@@ -6,6 +6,158 @@
 
 import numpy as np
 
+from mo.utils.error import Error
+
+dynamic_dimension = np.ma.masked
+# numpy masked array for integer values forces us to select one integer number to be considered as a missing/invalid
+# value. Since the primary purpose of usage of masked arrays in the MO is to specify dynamic dimension the big prime
+# (by modulo) negative number is selected as such a value
+dynamic_dimension_value = -1000000007
+
+
+def shape_array(value, dtype=np.int64):
+    # if the input already have masked values then we need to explicitly convert to dynamic_dimension_value and create
+    # masked array from scratch, because otherwise method masked_equal will convert masked elements to "nan" value
+    if isinstance(value, Iterable) and (not isinstance(value, np.ndarray) or value.ndim != 0):
+        value = [item if item is not dynamic_dimension else dynamic_dimension_value for item in value]
+    return np.ma.masked_equal(value, dynamic_dimension_value).astype(dtype=dtype)
+
+
+def compare_dimensions(dim1, dim2):
+    """
```

Comment:
`are_dims_compatible` seems more logical name here.

Reply:
Renamed. I do not like to use `are` and `is` in the name so, keep it as `compatible_dims`. It looks quite good in the code, for example:

```python
    if compatible_dims(d1, d2): 
        # do something
```